From 12f5f06e8855b653c98b75de55a45098bb468d57 Mon Sep 17 00:00:00 2001
From: Daniel Lange <DLange@git.local>
Date: Mon, 1 Feb 2021 16:02:31 +0100
Subject: [PATCH] Check for sortTimeout to not run towards -inf

Seems to happen on Mac OS "Big Sur" (~forced application sleep)
Partial fix for #510
---
 ScreenManager.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ScreenManager.c b/ScreenManager.c
index 64af535b..df5fed4f 100644
--- a/ScreenManager.c
+++ b/ScreenManager.c
@@ -202,7 +202,8 @@ void ScreenManager_run(ScreenManager* this, Panel** lastFocus, int* lastKey) {
          }
       }
       if (ch == ERR) {
-         sortTimeout--;
+         if (sortTimeout > 0)
+            sortTimeout--;
          if (prevCh == ch && !timedOut) {
             closeTimeout++;
             if (closeTimeout == 100) {
